frontend/src/contexts/AppContext.tsx

```diff
@@ -18,25 +26,60 @@ interface State {
   conferenceStatus: IConferenceStatus;
 }
 
-const reducer = (state, action) => {
-  const allowedActions = [
-    'FISHBOWL_STARTED',
-    'FISHBOWL_READY',
-    'FISHBOWL_STATUS',
-    'FISHBOWL_ENDED',
-    'JOIN_GUEST',
-    'JOIN_USER'
-  ];
-  const { type, ...actionData } = action;
+type fishbowlStartedAction = {
+  type: typeof FISHBOWL_STARTED;
+  fishbowlStarted: boolean;
+};
+
+type fishbowlReadyAction = {
+  type: typeof FISHBOWL_READY;
+  fishbowlReady: boolean;
+};
 
-  if (allowedActions.includes(type)) {
-    return {
-      ...state,
-      ...actionData
-    };
-  }
+type fishbowlStatusAction = {
+  type: typeof FISHBOWL_STATUS;
+  fishbowlReady?: boolean;
+  fishbowlStarted?: boolean;
+  isGuest?: boolean;
+  prejoin?: boolean;
+  conferenceStatus: IConferenceStatus;
+};
 
-  return state;
+type fishbowlEndedAction = {
+  type: typeof FISHBOWL_ENDED;
+  conferenceStatus: IConferenceStatus;
+};
+
+type joinGuestAction = {
+  type: typeof JOIN_GUEST;
+  isGuest: boolean;
+  prejoin: boolean;
+};
+
+type joinUserAction = {
+  type: typeof JOIN_USER;
+  prejoin?: boolean;
+  isGuest?: boolean;
+};
+
+type Actions =
+  | fishbowlStartedAction
+  | fishbowlReadyAction
+  | fishbowlStatusAction
+  | fishbowlEndedAction
+  | joinGuestAction
+  | joinUserAction;
+
+type FooReducer = Reducer<State, Actions>;
```

Comment:
Maybe you forgot to change the variable name :+1:

Reply:
Don't you like a good FooReducer?

Reply:
Yes, I forgot 😓

Reply:
Fixed.